### PR TITLE
DDF-2132 Added version properties for karaf dependencies

### PR DESCRIPTION
**Pull request by emmberk**
#### What does this PR do?
This PR adds `${karaf.version}` to karaf dependencies so that the version does not rely on dependency management (see [DDF-2985](https://codice.atlassian.net/browse/DDF-2985)). This PR shouldn't affect any versions. It just cleans up the dependency management before the Karaf upgrade to 4.1.1.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@vinamartin @mcalcote @josephthweatt 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@beyelerb
@clockard
#### How should this be tested? (List steps with links to updated documentation)
Full build.
#### What are the relevant tickets?
[DDF-2132](https://codice.atlassian.net/browse/DDF-2132)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by emmberk** [comment]
build now

File: catalog/resourcemanagement/resourcemanagement-app/pom.xml [hunk 1]
```diff
@@ -80,7 +80,6 @@
             <plugin>
                 <groupId>org.apache.karaf.tooling</groupId>
                 <artifactId>karaf-maven-plugin</artifactId>
-                <version>${karaf.version}</version>
```
Comment (josephthweatt) [inline catalog/resourcemanagement/resourcemanagement-app/pom.xml]: Just curious, why did you delete this line but add it everywhere else?

Comment (emmberk) [inline catalog/resourcemanagement/resourcemanagement-app/pom.xml]: The version for the `karaf-mavin-plugin` is defined in the`pluginManagement` section of the root pom: https://github.com/codice/ddf/blob/master/pom.xml#L396. I wasn't sure what the beast practices are when it comes to `pluginManagement` (as opposed to `dependencyManagement`), so I just made these `karaf-mavin-plugin` lines consistent with the rest of the poms in DDF.

**Comment by emmberk** [comment]
build now
